docs/blog/2023-03-31-zephyr-3-2.md

```diff
@@ -0,0 +1,303 @@
+---
+title: "Zephyr 3.2 Update"
+author: Pete Johanson
+author_title: Project Creator
+author_url: https://gitlab.com/petejohanson
+author_image_url: https://www.gravatar.com/avatar/2001ceff7e9dc753cf96fcb2e6f41110
+tags: [firmware, zephyr, core]
+---
+
+I'm happy to announce that we have completed the [work](https://github.com/zmkfirmware/zmk/pull/1499) to upgrade ZMK to [Zephyr 3.2](https://docs.zephyrproject.org/3.2.0/releases/release-notes-3.2.html)!
+
+[petejohanson] did the upgrade work to adjust ZMK for the Zephyr changes, with help from [Nicell] on the LVGL pieces.
+
+- Upgrade to LVGL 8.x API, and move to the new Kconfig settings.
+- Tons of RP2040 work.
+- Zephyr core API changes, including DTS `label` use changes.
+- Move to [pinctrl](https://docs.zephyrproject.org/3.2.0/hardware/pinctrl/index.html) Zephyr subsystem.
+
+## Getting The Changes
+
+Use the following steps to update to the latest tooling in order to properly use the new ZMK changes:
+
+### User Config Repositories Using GitHub Actions
+
+Existing user config repositories using Github Actions to build will pull down Zephyr 3.2 automatically, however if you created your user config a while ago, you may need to update it to reference our shared build configuration to leverage the correct Docker image.
+
+1.  Replace the contents of your `.github/workflows/build.yaml` with:
+
+    ```
+    on: [push, pull_request, workflow_dispatch]
+
+    jobs:
+        build:
+        uses: zmkfirmware/zmk/.github/workflows/build-user-config.yml@main
+    ```
+
+1.  Add a new file to your repository named `build.yaml`:
```

Comment:
```suggestion
1.  If it doesn't exist already, add a new file to your repository named `build.yaml`:
```
Might be worth qualifying since there was a period of time after 3.0 upgrade that we had build.yaml but not the reusable workflow